Source/WebKit/UIProcess/Inspector/mac/WebInspectorUIProxyMac.mm

```diff
@@ -823,6 +823,36 @@ - (IBAction)_popUpButtonAction:(id)sender
     [inspectedView setFrame:inspectedViewFrame];
 }
 
+static RetainPtr<NSArray> transferConstraintsAwayFromInspectedView(NSView *oldConstraintView, NSView *inspectedView, NSView *newContainerView)
+{
+    RetainPtr transferredConstraints = adoptNS([[NSMutableArray alloc] init]);
+    if (oldConstraintView != nil) {
+        for (NSLayoutConstraint *oldConstraint in oldConstraintView.constraints) {
+            // Ignore custom subclasses of NSLayoutConstraint, because they may use custom attribute
+            // values which fail when trying to initialize a new constraint.
+            if ([oldConstraint isMemberOfClass:[NSLayoutConstraint class]]) {
+                // Update the first or second item to refer to container view instead
+                BOOL inspectedIsFirst = (oldConstraint.firstItem == inspectedView);
+                BOOL inspectedIsSecond = (oldConstraint.secondItem == inspectedView);
+
+                if (inspectedIsFirst || inspectedIsSecond) {
+                    NSObject *newFirstItem = inspectedIsFirst ? newContainerView : oldConstraint.firstItem;
+                    NSObject *newSecondItem = inspectedIsSecond ? newContainerView : oldConstraint.secondItem;
+
+                    NSLayoutConstraint *newConstraint = [NSLayoutConstraint constraintWithItem:newFirstItem attribute:oldConstraint.firstAttribute relatedBy:oldConstraint.relation toItem:newSecondItem attribute:oldConstraint.secondAttribute multiplier:oldConstraint.multiplier constant:oldConstraint.constant];
+                    [transferredConstraints.get() addObject:newConstraint];
+                    oldConstraint.active = NO;
```

Comment:
What dis do?

Reply:
Perhaps the question here is about the same gist as whether the constraints are copied or transferred? What this is doing in creating a new constraint that has all the same properties as the previous constraint, except for the related items (the views that the constraint will affect the layout of). The new constraint is returned to be added to the intermediate view that is created to manage the inspector and inspected views. The `oldConstraint.active = NO` is a conventional way of disabling a constraint which involves removing it from the view it's attached to.